doc.md

```diff
@@ -1085,6 +1081,35 @@ PATCH /api/users/12
 ('flair: null' removes the flair.)
 ```
 
+<a name='give-user-role'></a>
+### Give a role to a user [POST /api/users/:userID/roles]
++ requires [permission](#permissions): `manageRoles`
+  + also require more permissions (see below)
++ **in-url** userID (ID) - The ID of the user to give the role to
++ roleID (ID) - The ID of the role to be given
+
+The role may only be given if the requesting user has all the permissions that are specified by that role (even if the value specified by the role is false).
+
+On success, emits [user/update](#user-update) and returns `{}`.
+
+<a name='take-user-role'></a>
+### Take a role from a user [DELETE /api/users/:userID/roles/:roleID]
++ requires [permission](#permissions): `manageRoles`
+  + also requires more permissions (see below)
++ **in-url** userID (ID) - The ID of the user to take the role from
++ **in-url** roleID (ID) - The ID of the role to be taken
+
+As with [giving a role](#give-user-role), the role may only be taken if the requesting user has all the permissions that are specified by that role (even if the value specified by the role is false).
+
+On success, emits [user/update](#user-update) and returns `{}`.
```

Comment:
imo these should return the **new** `roleIDs` value.

Reply:
Other than that, LGTM!

Reply:
I considered it yeah, but like you said, inconsistency. It can be assumed that if the request was successful, then new roleIDs = old roleIDs minus removed role.